wordpress-mobile/WordPress-Android
# [Modules Restructuring] Restructure Networking Module

Parent #16562
Fixes #16567

This PR restructures the `networking` module by disassociating the [WordPress-Networking-Android](https://github.com/wordpress-mobile/WordPress-Networking-Android) repo and removing the extra `WordPressNetworking` level from it. This effectively makes `networking` a pure module, without any extra inner levels or unnecessary build files lying around (like `settings.gradle`, `.gitignore`, etc).

PS: @ashiagr I added you as the main reviewer, that is, in addition to the @wordpress-mobile/owl-team team itself, since I know you were working with Reader and Discover in the past, which will come handy for testing purposes. As such, I think you might be the best person to sign-off on that change for WPAndroid. 🥇 

-----

Question (❓): As per my licensing question on this other related PR of mine [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16744) and since I don't see any `LICENCE` file for this module, like I did for the `analytics` mode, which is pointing to `MIT`, I am not sure if we should include a licence for this module too or exclude the licensing information from the `analytics` and all other modules in general so that we start depending solely on the root level `LICENCE` that WPAndroid is using, which is `GPLv2`, wdyt? 🤔 

-----

@oguzkocer related to your comments on this other related PR of mine [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16744) I am still waiting for us to align on both, the module [naming convention](https://github.com/wordpress-mobile/WordPress-Android/pull/16744#discussion_r897678064) and the [README file](https://github.com/wordpress-mobile/WordPress-Android/pull/16744#discussion_r897054484). Then, I'll proceed and make the necessary update(s), if any, on all the PRs. You will see me progressing with my PRs anyway just so that I kepp move forward, but that doesn't mean anything, I can and will make the change(s) we end up agreeing on afterwards and before I merge any of my PRs so don't worry if you see me creating more PRs like that.

-----

To test:
- Build the app and install it on your device or emulator.
- Go to `Reader` -> `DISCOVER` tab.
- Trigger pull-to-refresh and verify that the tab is updated with new content (see `ReaderDiscoverLogic.kt` class, line `123` and the `WordPress.getRestClientUtilsV2(...)` call.

The above is just one example of how this change can tested so that we could quickly verify the overall correctness of this PR. However, feel free to smoke test other aspects as well, especially if you are aware any or all the other places where `networking` is being used to fetch data instead of [FluxC](https://github.com/wordpress-mobile/WordPress-FluxC-Android).

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
